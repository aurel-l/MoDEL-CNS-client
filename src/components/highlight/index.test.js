import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Highlight from '.';

describe('<Highlight />', () => {
  it('renders without crashing', () => {
    expect(toJSON(shallow(<Highlight />))).toBe('');
    expect(toJSON(shallow(<Highlight highlight="h" />))).toBe('');
    expect(toJSON(shallow(<Highlight>hello</Highlight>))).toBe('hello');
    expect(toJSON(shallow(<Highlight highlight="h">{''}</Highlight>))).toBe('');
    expect(toJSON(shallow(<Highlight highlight="">hello</Highlight>))).toBe(
      'hello',
    );
    expect(
      toJSON(
        shallow(
          <Highlight>
            <span>Hello</span>
          </Highlight>,
        ),
      ),
    ).toMatchSnapshot();
  });

  it('renders with highlight', () => {
    expect(
      shallow(<Highlight highlight="ll">hello</Highlight>),
    ).toMatchSnapshot();
    expect(
      shallow(<Highlight highlight="he">hello hello</Highlight>),
    ).toMatchSnapshot();
    expect(
      shallow(
        <Highlight highlight="he">
          <span>hello</span>
        </Highlight>,
      ),
    ).toMatchSnapshot();
    expect(
      shallow(<Highlight highlight="2">{1234}</Highlight>),
    ).toMatchSnapshot();
    expect(
      shallow(<Highlight highlight={2}>{1234}</Highlight>),
    ).toMatchSnapshot();
  });
});
