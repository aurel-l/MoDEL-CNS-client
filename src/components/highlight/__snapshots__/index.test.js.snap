// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> renders with highlight 1`] = `
<span
  className="highlightable"
>
  <span
    key="0"
  >
    he
  </span>
  <mark
    key="1"
  >
    ll
  </mark>
  <span
    key="2"
  >
    o
  </span>
</span>
`;

exports[`<Highlight /> renders with highlight 2`] = `
<span
  className="highlightable"
>
  <mark
    key="0"
  >
    he
  </mark>
  <span
    key="1"
  >
    llo 
  </span>
  <mark
    key="2"
  >
    he
  </mark>
  <span
    key="3"
  >
    llo
  </span>
</span>
`;

exports[`<Highlight /> renders with highlight 3`] = `
<span>
  hello
</span>
`;

exports[`<Highlight /> renders with highlight 4`] = `
<span
  className="highlightable"
>
  <span
    key="0"
  >
    1
  </span>
  <mark
    key="1"
  >
    2
  </mark>
  <span
    key="2"
  >
    34
  </span>
</span>
`;

exports[`<Highlight /> renders with highlight 5`] = `
<span
  className="highlightable"
>
  <span
    key="0"
  >
    1
  </span>
  <mark
    key="1"
  >
    2
  </mark>
  <span
    key="2"
  >
    34
  </span>
</span>
`;

exports[`<Highlight /> renders without crashing 1`] = `
<span>
  Hello
</span>
`;
